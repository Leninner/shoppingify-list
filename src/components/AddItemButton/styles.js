import styled from 'styled-components';

export const StyledAddItemContainer = styled.div`
  position: absolute;
  width: 100%;
  height: 8.186rem;
  background-color: white;
  right: 0;
  bottom: 0;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const SimilarStyles = `
position: relative;
width: 19.368rem;
display: flex;
align-items: center;
justify-content: center;
`;

export const Div = styled.div`
  ${SimilarStyles}
`;

export const Form = styled.form`
  ${SimilarStyles}
`;

export const BoxInput = styled.input`
  width: 100%;
  height: 3.828rem;
  border: 2px solid #f9a109;
  border-radius: 12px;
  outline: none;
  padding: 0 1.102rem;

  &::placeholder {
    font-style: normal;
    font-weight: 500;
    font-size: 14px;
    line-height: 17px;
    color: #bdbdbd;
  }

  &:disabled {
    border: 2px solid #c1c1c4;
  }
`;

export const ButtonAddItem = styled.button`
  position: absolute;
  width: 5.459rem;
  height: 3.828rem;
  background: #f9a109;
  border-radius: 12px;
  border: none;
  right: 0;
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  line-height: 20px;
  color: #ffffff;

  &:disabled {
    background: #c1c1c4;
  }
`;

export const ButtonConfirm = styled.button`
  min-width: 87.35px;
  padding: 0 2vh;
  height: 61.25px;
  border-radius: 12px;
  border: none;
  background: transparent;
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  line-height: 20px;
  transition: all 0.2s ease 0s;

  ${({ isCancel }) =>
    isCancel
      ? 'color: #34333A; margin-right: 10px; background: rgb(227, 227, 227); &:hover { background: rgb(221, 221, 221); }'
      : 'background: #F9A109;  color: #FFFFFF; margin-right: 10px;'}
`;

export const ButtonCompletingList = styled.button`
  min-width: 87.35px;
  padding: 0 2vh;
  height: 61.25px;
  border-radius: 12px;
  border: none;
  background: transparent;
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  line-height: 20px;
  transition: all 0.2s ease 0s;

  ${({ isCancelList }) =>
    isCancelList
      ? 'color: #34333A; margin-right: 10px; background: rgb(227, 227, 227); &:hover { background: rgb(221, 221, 221); }'
      : 'background: #56CCF2; border-radius: 12px; color: #FFFFFF;'}
`;
