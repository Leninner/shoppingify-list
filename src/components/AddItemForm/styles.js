import styled from 'styled-components';

const mainStyles = `
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
`;

export const StyledAddItemForm = styled.form`
  width: 100%;
  height: 100%;
  background-color: white;
  padding-top: 20px;

  h3 {
    margin-left: 20px;
    font-style: normal;
    font-weight: 500;
    font-size: 24px;
    line-height: 30px;
    color: #000000;
  }
`;

export const Campo = styled.label`
  ${mainStyles}
  color: #34333a;
  display: flex;
  flex-direction: column;
  margin: 18px 0;

  &:focus-within > span {
    color: #f9a109;
  }
`;

export const Entrada = styled.input`
  width: 309.89px;
  height: 61.25px;
  border: 2px solid #bdbdbd;
  border-radius: 12px;
  outline: none;
  padding: 0px 16px;
  margin-top: 6.08px;
  color: #000000;
  ${mainStyles}

  &::placeholder {
    font-style: normal;
    ${mainStyles}
    color: #bdbdbd;
  }

  &:focus {
    border: 2px solid #f9a109;
  }
`;

export const EntradaNote = styled.textarea`
  width: 309.89px;
  height: 100px;
  border: 2px solid #bdbdbd;
  border-radius: 12px;
  outline: none;
  resize: vertical;
  max-height: 150px;
  padding: 10px 16px;
  margin-top: 10px;

  &::placeholder {
    ${mainStyles}
    color: #bdbdbd;
  }

  &:focus {
    border: 2px solid #f9a109;
  }
`;

export const Section = styled.section`
  width: 100%;
  height: 75%;
  overflow-y: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
`;
