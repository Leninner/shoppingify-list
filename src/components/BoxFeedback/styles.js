import styled from 'styled-components';

export const StyledBoxFeedback = styled.div`
  position: absolute;
  width: 19.298rem;
  height: 8.119rem;
  top: 43.7px;
  left: calc(50% - 39.69%);
  background: #80485b;
  border-radius: 24px;
`;

export const Botella = styled.img`
  position: absolute;
  top: -20px;
  left: 10px;
`;

export const Info = styled.div`
  position: absolute;
  right: 0;
  width: 165px;
  top: 17.61px;
  right: 27.59px;
`;

export const Description = styled.p`
  font-weight: bold;
  font-size: 16px;
  line-height: 20px;
  color: #ffffff;
  letter-spacing: 0.5px;
`;

export const Button = styled.button`
  margin-top: 13.61px;
  width: 7.5rem;
  height: 2.5rem;
  background: #ffffff;
  box-shadow: 0px 2px 12px rgba(0, 0, 0, 0.04);
  border-radius: 12px;
  border: none;
  font-style: normal;
  font-weight: bold;
  font-size: 14px;
  line-height: 17px;
  color: #34333a;
`;
