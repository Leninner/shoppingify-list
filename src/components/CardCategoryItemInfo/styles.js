import styled from 'styled-components';

export const StyledCardCategoryItemInfo = styled.div`
  width: 100%;
  height: 100%;
  padding: 27px 30px;
  overflow: auto;
  padding-bottom: 150px;
  background-color: #fff;
`;

export const Return = styled.div`
  display: flex;
  align-items: center;
  color: #f9a109;
  margin-bottom: 20px;

  span {
    margin-right: 6.54px;
    text-decoration: none;
  }

  &:hover {
    color: #f9a109;
    text-decoration: underline;
  }
`;

export const ImgItem = styled.img`
  width: 100%;
  height: 13.731rem;
  border-radius: 25px;
  object-fit: cover;
`;

export const ContainerInfo = styled.div`
  margin-top: 20px;

  span {
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 15px;
    color: #c1c1c4;
  }

  h3 {
    font-style: normal;
    font-weight: 500;
    font-size: 24px;
    line-height: 30px;
    color: #000000;
  }

  p {
    font-style: normal;
    font-weight: 500;
    font-size: 18px;
    line-height: 22px;
    color: #000000;
  }
`;
