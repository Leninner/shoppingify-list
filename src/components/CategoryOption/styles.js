import styled from 'styled-components';

export const Option = styled.div`
  width: 294.36px;
  height: 46.79px;
  border-radius: 12px;
  font-weight: 500;
  font-size: 18px;
  line-height: 22px;
  margin: 2px 0;
  display: flex;
  align-items: center;
  padding: 0px 17.63px;
  color: #828282;

  &:hover {
    background: #f2f2f2;
    color: #34333a;
  }
`;
