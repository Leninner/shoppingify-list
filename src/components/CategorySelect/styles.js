import styled from 'styled-components';

export const ContainerOptions = styled.div`
  margin-top: 10px;
  width: 309.89px;
  height: 168.38px;
  background: #ffffff;
  border: 1px solid #e0e0e0;
  box-sizing: border-box;
  box-shadow: 0px 2px 12px rgba(0, 0, 0, 0.04);
  border-radius: 12px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  padding: 10px 0;
`;
