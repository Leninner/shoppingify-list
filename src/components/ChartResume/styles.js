import styled from 'styled-components';

export const StyledChartResume = styled.div`
  width: 100%;
  padding: 0 5%;
  height: 100%;
  margin-bottom: 150px;

  h1 {
    margin: 45px 0;
    font-style: normal;
    font-weight: 500;
    font-size: 24px;
    line-height: 30px;
    color: #000000;
  }
`;
