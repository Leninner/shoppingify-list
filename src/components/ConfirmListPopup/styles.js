import styled from 'styled-components';

export const ConfirmListPopupWrapper = styled.section`
  width: 100%;
  height: 100vh;
  background: rgba(0, 0, 0, 0.1);
  position: absolute;
  top: 0;
  left: 0;
  z-index: 100;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const ConfirmListContainer = styled.div`
  position: relative;
  width: 508.52px;
  height: 219.39px;
  background: #ffffff;
  border-radius: 24px;

  h3 {
    position: absolute;
    width: 410.82px;
    height: 60px;
    top: 33.25px;
    left: 38.51px;
    font-style: normal;
    font-weight: 500;
    font-size: 24px;
    line-height: 30px;
    color: #34333a;
  }

  span {
    position: absolute;
    top: 1rem;
    right: 1.5rem;
    color: #828282;
  }

  div {
    position: absolute;
    bottom: 29.26px;
    right: 30.15px;
  }
`;

export const FeddbackButton = styled.button`
  width: 87.35px;
  height: 61.25px;
  border-radius: 12px;
  outline: none;
  border: none;
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  line-height: 20px;

  ${({ isCancel }) => (!isCancel ? ` background: #eb5757; color: white` : `background: white; color: #34333A`)}
`;
