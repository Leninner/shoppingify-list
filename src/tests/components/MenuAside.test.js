import { MenuAside } from '../../components/MenuAside';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { MenuItem } from '../../components/MenuAside/styles';

describe('Debe encontrar los elementos correctos en el DOM', () => {
  test('Debe encontrar el componente en el documento', () => {
    const menuAside = shallow(<MenuAside />);
    expect(menuAside).toMatchSnapshot();
  });

  test('Debe encontrar 3 elementos span en el DOM', () => {
    const menuAside = shallow(<MenuAside />);
    expect(menuAside.find(MenuItem).length).toBe(3);
  });
});
