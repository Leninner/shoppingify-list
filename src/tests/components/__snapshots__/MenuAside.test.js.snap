// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe encontrar los elementos correctos en el DOM Debe encontrar el componente en el documento 1`] = `
<styled.div>
  <LogoComponent />
  <styled.div>
    <styled.span
      className="material-icons"
      top={40.34}
    >
      
    </styled.span>
    <styled.span
      className="material-icons"
      top={49.59}
    >
      
    </styled.span>
    <styled.span
      className="material-icons"
      top={60.16}
    >
      
    </styled.span>
  </styled.div>
  <ShoppingCart
    WrapperComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": e {
          "baseHash": -1179868038,
          "baseStyle": undefined,
          "componentId": "sc-dkPtRN",
          "isStatic": false,
          "rules": Array [
            "
  position: absolute;
  left: 35.83%;
  top: ",
            [Function],
            ";
  width: 20.04px;
  height: 16.25px;
  color: #454545;
",
          ],
          "staticRulesId": "",
        },
        "foldedComponentIds": Array [],
        "render": [Function],
        "shouldForwardProp": undefined,
        "styledComponentId": "sc-dkPtRN",
        "target": "span",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
  />
</styled.div>
`;
